crates/control/src/calibration_controller.rs

```diff
@@ -0,0 +1,299 @@
+            .initial_to_calibration_stabilization_delay;
+        match self.current_calibration_command {
+            CalibrationCommand::INACTIVE => {
+                if primary_state_transition_to_calibration {
+                    info!(
+                        "Calibration is activated, waiting for {}s until the Robot is stable.",
+                        initial_stabilization_delay.as_secs_f32()
+                    );
+
+                    Some(CalibrationCommand::INITIALIZE {
+                        started_time: *current_cycle_time,
+                    })
+                } else {
+                    None
+                }
+            }
+            CalibrationCommand::INITIALIZE {
+                started_time: activated_time,
+            } => match context.primary_state {
+                PrimaryState::Calibration => {
+                    let waiting_duration = current_cycle_time
+                        .start_time
+                        .duration_since(activated_time.start_time)
+                        .unwrap_or_default();
+
+                    if waiting_duration >= initial_stabilization_delay {
+                        Some(self.try_transition_to_look_at(*context.cycle_time))
+                    } else {
+                        None
+                    }
+                }
+                _ => None,
+            },
+            CalibrationCommand::LOOKAT { dispatch_time, .. } => {
+                let time_diff = current_cycle_time
+                    .start_time
+                    .duration_since(dispatch_time.start_time)
+                    .unwrap_or_default();
+
+                if time_diff > look_at_dispatch_waiting {
+                    Some(CalibrationCommand::CAPTURE {
+                        dispatch_time: *current_cycle_time,
+                    })
+                } else {
+                    None
+                }
+            }
+            CalibrationCommand::CAPTURE { dispatch_time } => {
+                // TODO decide a better way to handle the two cameras. Perhaps set a capture position like in the look-at command?
+                let (values_top, goto_next_position_top) =
+                    collect_filtered_values(&context.measurement_top, &dispatch_time);
+                let (values_bottom, goto_next_position_bottom) =
+                    collect_filtered_values(&context.measurement_bottom, &dispatch_time);
+
+                if let Some(measurement) = values_top {
+                    self.current_measurements.push(measurement);
+                }
+                if let Some(measurement) = values_bottom {
+                    self.current_measurements.push(measurement);
+                }
+
+                if goto_next_position_top || goto_next_position_bottom {
+                    Some(self.try_transition_to_look_at(*context.cycle_time))
+                } else {
+                    None
+                }
+            }
+            CalibrationCommand::PROCESS => {
```

Comment:
Why do you need this state?
Can't you directly do the processing when there are no more targets to look at?

Reply:
In principle we can and will simplify things at a glance. But, the function that gets the next available look-at is called in two places (initialise-> look-at), and (capture -> look-at). So it was a bit more obvious to see where the processing happens by having a seperate state.

The drawback is that the processing starts one cycle later.

Reply:
```
    fn get_next_look_at_or_processing(
        &mut self,
        dispatch_time: CycleTime,
    ) -> Option<CalibrationState> {
        self.get_next_look_at()
            .map(|(target, camera)| CalibrationState::LookAt {
                camera,
                target,
                dispatch_time,
            })
    }
```